File: C++/Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n <= 0)
            return "";
        int i, j;
        string last = "";
        string here = "1";
        while (--n) {
            last = here;
            here = "";
            for (i=0; i<last.length(); i++) {
                for (j=i; j<last.length()-1; j++) {
                    if (last[j] != last[j+1]) {
                        here += j - i + '1';
                        here += last[j];
                        break;
                    }
                }
                if (j == last.length() - 1) {
                    here += j - i + '1';
                    here += last[j];
                }
                i = j;
            }
        }
        return here;
    }
};

File: Java/Count and Say.java
public class Solution {
	public String countAndSay(int n) {
		String t, r = "1";
		int l;
		for (int i = 1; i < n; i++) {
			l = r.length();
			t = r;
			r = "";
			int no = 1;
			for (int j = 1; j < l; j++)
				if (t.charAt(j) == t.charAt(j - 1)) {
					no++;
				} else {
					r += "" + no + t.charAt(j - 1);
					no = 1;
				}
			r += "" + no + t.charAt(l - 1);
		}
		return r;
	}
}